benchmark.results

```diff
@@ -1,649 +1,649 @@
+CustomDebugString: BinaryTree (reference)      33717.0 ns ± 134.47 %        41471
+CustomDebugString: Color (generic)              1305.0 ns ±   9.15 %      1000000
+CustomDebugString: Color (reference)            1046.0 ns ±   9.48 %      1000000
+CustomDebugString: ASCII (generic)              1397.0 ns ±   9.33 %      1000000
+CustomDebugString: ASCII (reference)            1074.0 ns ± 800.78 %      1000000
+CustomEquatable: Point1 (generic)                116.0 ns ±  29.40 %      1000000
+CustomEquatable: Point1 (reference)               23.0 ns ±  68.91 %      1000000
+CustomEquatable: Point2 (generic)                177.0 ns ±  25.80 %      1000000
+CustomEquatable: Point2 (reference)               23.0 ns ±  65.61 %      1000000
+CustomEquatable: Point3 (generic)                248.0 ns ±  20.37 %      1000000
+CustomEquatable: Point3 (reference)               23.0 ns ± 105.10 %      1000000
+CustomEquatable: Point4 (generic)                320.0 ns ±  14.79 %      1000000
+CustomEquatable: Point4 (reference)               23.0 ns ±  34.77 %      1000000
+CustomEquatable: Point5 (generic)                393.0 ns ±  22.47 %      1000000
+CustomEquatable: Point5 (reference)               23.0 ns ±  91.79 %      1000000
+CustomEquatable: Point6 (generic)                476.0 ns ±  14.84 %      1000000
+CustomEquatable: Point6 (reference)               23.0 ns ± 179.40 %      1000000
+CustomEquatable: Point7 (generic)                545.0 ns ±  14.36 %      1000000
+CustomEquatable: Point7 (reference)               25.0 ns ±  23.52 %      1000000
+CustomEquatable: Point8 (generic)                637.0 ns ±  12.97 %      1000000
+CustomEquatable: Point8 (reference)               24.0 ns ±  39.64 %      1000000
+CustomEquatable: Point9 (generic)                724.0 ns ±  10.41 %      1000000
+CustomEquatable: Point9 (reference)               25.0 ns ± 112.43 %      1000000
+CustomEquatable: Point10 (generic)               768.0 ns ±   9.23 %      1000000
+CustomEquatable: Point10 (reference)              25.0 ns ± 100.57 %      1000000
+CustomEquatable: Point11 (generic)               840.0 ns ±  87.71 %      1000000
+CustomEquatable: Point11 (reference)              25.0 ns ±  32.02 %      1000000
+CustomEquatable: Point12 (generic)               979.0 ns ±   9.50 %      1000000
+CustomEquatable: Point12 (reference)              25.0 ns ±  73.99 %      1000000
+CustomEquatable: Point13 (generic)              1000.0 ns ±  10.01 %      1000000
+CustomEquatable: Point13 (reference)              25.0 ns ±  51.44 %      1000000
+CustomEquatable: Point14 (generic)              1061.0 ns ±  10.12 %      1000000
+CustomEquatable: Point14 (reference)              25.0 ns ±  40.39 %      1000000
+CustomEquatable: Point15 (generic)              1145.0 ns ±  20.39 %      1000000
+CustomEquatable: Point15 (reference)              25.0 ns ±  38.51 %      1000000
+CustomEquatable: Point16 (generic)              1213.0 ns ±  12.65 %      1000000
```

Comment:
@saeta Runtime performance is degraded due to addition of KeyPaths. It's unclear how much effort will it take to bring it back down while preserving KeyPaths in the API.